sklearn/utils/estimator_checks.py

```diff
@@ -444,7 +445,7 @@ def _should_be_skipped_or_marked(estimator, check):
     return False, "placeholder reason that will never be used"
 
 
-def parametrize_with_checks(estimators, *, legacy=True):
+def parametrize_with_checks(estimators, *, legacy: bool = True):
```

Comment:
Any reason why we are introducing a type hint only here?

Reply:
We tend to put the typehints for trivial new parameters, I had forgotten this one.